web/soubory/program-nahled.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * @param obalNahledu jquery element set element, ve kterém je náhled
+ * @param obalProgramu jquery element set element, ve kterém je program (kvůli
+ *  úpravě šířky při zobrazení náhledu)
+ * @param odkazy jquery element set odkazy na aktivity. Musí mít nastavený data
+ *  atribut program-nahled-id. Nejde o všechny odkazy, ale jen odkazy na
+ *  aktivity.
+ * @param prepinac jquery element set prvek, kterým se zapíná/vypíná zobrazení
+ *  náhledu.
+ */
+function programNahled(obalNahledu, obalProgramu, odkazy, prepinac) {
+
+  class Nahled {
+    constructor(obalNahledu, obalProgramu, prepinac, storage) {
+      this.obalNahledu = obalNahledu
+      this.obalProgramu = obalProgramu
+      this.storage = storage
+      this.prepinac = prepinac
+
+      if (
+        this.viditelny || // ručně zapnuto
+        this.viditelny === null && $(document).width() > 1000 // nenastaveno
```

Comment:
Tohle přeskočí úvodní stav, kdy `this.viditelny` je undefined, to je schválně?

Reply:
Tohle by melo volat getter a ten vola localStorage.getItem, ktery myslim vraci null, pokud tam hodnota neni pritomna. Myslim, ze by to takto mohlo byt funkcni.

Reply:
Pravda